module/src/components/rating/rating.stories.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+
+import { Form } from '../..';
+import { StoryUtils } from '../../stories/storyUtils';
+import { IconUtils } from '../icon';
+import { Rating } from './rating.component';
+
+/** metadata */
+
+export default StoryUtils.createMeta(Rating, 'Form', 'Rating', {});
+
+/** component template */
+
+// const Template = StoryUtils.createTemplate(ScrollToEndListener as React.FC<IScrollToEndListenerProps<any, any>>);
+
+/** stories */
+
+export const Default = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} />
+    </div>
+  );
+};
+
+export const ReadOnly = () => {
+  return (
+    <div>
+      <Rating value={3} />
+    </div>
+  );
+};
+
+export const Half = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} step={0.5} />
+    </div>
+  );
+};
+
+export const CustomMax = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} maximum={10} />
+    </div>
+  );
+};
+
+export const CustomIcons = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating
+        bind={formProp('rating').bind()}
+        emptyIcon={IconUtils.getIconDefinition('Icomoon', 'airplane')}
+        filledIcon={IconUtils.getIconDefinition('Icomoon', 'airplane')}
+      />
+    </div>
+  );
+};
+export const CustomDOM = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} emptyIcon={<p>X</p>} filledIcon={<p>X</p>} />
+    </div>
+  );
+};
+export const CustomDOMFromIndex = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} emptyIcon={(index) => <p>{index + 1}</p>} filledIcon={(index) => <p>{index + 1}</p>} />
+    </div>
+  );
+};
+export const WithStatusAndIcons = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} maximum={10} leftIcon={IconUtils.getIconDefinition('Icomoon', 'planet')} pending error />
+    </div>
+  );
+};
+
+export const Radio = () => {
+  const { formProp } = Form.use({ rating: 0 });
+
+  return (
+    <div>
+      <Rating bind={formProp('rating').bind()} mode="radio" />
```

Comment:
On safari the styling for the radio button goofs a little. I have been hit by this before and if I recall the fix is -webkit-appearance none
![Screenshot 2022-03-30 at 09 36 53](https://user-images.githubusercontent.com/31133181/160789099-d515d1e5-532b-4e59-82ce-f693dd6c4111.png)